File: scrumote-backend/src/main/java/com/adach/scrumote/entity/SystemFeature.java
package com.adach.scrumote.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "system_feature_t")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true, onlyExplicitlyIncluded = true)
public class SystemFeature extends AbstractEntity {

  //region Data
  @Column(nullable = false, unique = true, length = 32)
  @NotNull
  @Size(max = 32)
  private String code;

  @Column(nullable = false)
  @NotNull
  private boolean enabled;
  //endregion
}

File: scrumote-frontend/src/app/_models/system-feature.ts
export class SystemFeature {

  id: number | null;
  version: number | null;

  code: string;
  enabled: boolean;

  constructor(id: number | null, version: number | null, code: string, enabled: boolean) {
    this.id = id;
    this.version = version;
    this.code = code;
    this.enabled = enabled;
  }

  static overwrite(to: SystemFeature, from: SystemFeature) {
    to.id = from.id;
    to.version = from.version;
    to.code = from.code;
    to.enabled = from.enabled;
  }
}
